### DISCUSSION TITLE
buttons to switch the waveform

### DISCUSSION THREAD

**Discussion by tiretrak** [marcel-licence/esp32_basic_synth#31] [Ideas]
is it possible to add buttons to switch the waveform(like 74hc165 or so)?

**Comment by marcel-licence** [comment]
I will take a look soon at the MCP23S17. This may help to use multiple buttons with less GPIO's. But you can also use existing GPIO's to change parameters.

**Reply by MichaelPNolan** [reply]
I did that - i have a button linked to the ESP for the wave form. I have another that rotates through the parameters of the synth and takes a single pot to adjust. Nice for a test - I went off and built 8 pots and a n ADC Mux and sent midi of those because it was too slow to work with one pot.

**Reply by marcel-licence** [reply]
I used finally the MCP23017 and MCP23S17 here: https://youtu.be/juwm6o3j8XA
So lots of options to control your settings.

**Reply by MichaelPNolan** [reply]
Cool montage. Yes, i could use the port expander(s) for my midi controller toy. Now I got to figure out how to send back midi messages to the controller that it can display in some fashion.

**Reply by MichaelPNolan** [reply]
I ordered those parts though I'm trying to put it into the midi-controller side.

**Reply by marcel-licence** [reply]
Sounds great =)
